src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -2281,6 +2512,134 @@ public function is_tag_closer() {
 		);
 	}
 
+	/**
+	 * Indicates the kind of matched token, if any.
+	 *
+	 * This differs from `get_token_name()` in that it always
+	 * returns a static string indicating the type, whereas
+	 * `get_token_name()` may return values derived from the
+	 * token itself, such as a tag name or processing
+	 * instruction tag.
+	 *
+	 * Possible values:
+	 *  - `#tag` when matched on a tag.
+	 *  - `#text` when matched on a text node.
+	 *  - `#cdata-section` when matched on a CDATA node.
+	 *  - `#comment` when matched on a comment.
+	 *  - `#presumptuous-tag` when matched on an empty tag closer.
+	 *  - `#funky-comment` when matched on a funky comment.
```

Comment:
`#doctype` seems to be missing here, although that may be a mistake…

Reply:
`#doctype` seems to be found in 9f2992070ebbbb8970c96c115da2fb74a7ebfda4